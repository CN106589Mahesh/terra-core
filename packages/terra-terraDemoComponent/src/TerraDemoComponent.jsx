import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import classNamesBind from 'classnames/bind';
import ThemeContext from 'terra-theme-context';
import styles from './TerraDemoComponent.module.scss';

const cx = classNamesBind.bind(styles);

const propTypes = {
  /**
   * Content to be displayed as the name
   */
  name: PropTypes.string,
};

const defaultProps = {
  name: 'default',
};

const TerraDemoComponent = ({ name, ...customProps }) => {
  const theme = React.useContext(ThemeContext);
  const TerraDemoComponentClassNames = classNames(
    cx([
      'terra-demo-component',
      theme.className,
    ]),
    customProps.className,
  );

  return (<button type="button" {...customProps} className={TerraDemoComponentClassNames}>{name}</button>);
};

TerraDemoComponent.propTypes = propTypes;
TerraDemoComponent.defaultProps = defaultProps;

export default TerraDemoComponent;
