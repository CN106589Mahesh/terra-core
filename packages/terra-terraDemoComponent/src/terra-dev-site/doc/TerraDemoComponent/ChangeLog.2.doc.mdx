import { Badge } from 'terra-terraDemoComponent/package.json?dev-site-package';
import ChangeLog from 'terra-terraDemoComponent/CHANGELOG.md';

<Badge />

<ChangeLog />
