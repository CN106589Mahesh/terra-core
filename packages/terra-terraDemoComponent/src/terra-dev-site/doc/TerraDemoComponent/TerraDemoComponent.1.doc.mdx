import { Badge } from 'terra-terraDemoComponent/package.json?dev-site-package';

import DefaultTerraDemoComponent from '../example/DefaultTerraDemoComponent?dev-site-example';

import PropsTable from 'terra-terraDemoComponent/src/TerraDemoComponent?dev-site-props-table';

<Badge />

# Terra Terrademocomponent

{insert description}

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-terraDemoComponent`

## Usage

```jsx
import React from 'react';
import TerraDemoComponent from 'terra-terraDemoComponent';

<TerraDemoComponent {props...} />
```

## Component Features

* [Cross-Browser Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#cross-browser-support)
* [LTR/RTL Support](https://engineering.cerner.com/terra-ui/contributing/terra-ui/component-standards#ltr--rtl-support)

## Examples
<DefaultTerraDemoComponent />

## Divider Props
<PropsTable />
