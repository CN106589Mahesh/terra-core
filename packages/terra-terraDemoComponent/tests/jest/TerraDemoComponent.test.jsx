import React from 'react';
import TerraDemoComponent from '../../src/TerraDemoComponent';

describe('TerraDemoComponent', () => {
  const defaultRender = <TerraDemoComponent />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should use the default value when no value is given', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.find('.terraDemoComponent').text()).toEqual('default');
  });

  // Structure Tests
  it('should have the class terraDemoComponent', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('terraDemoComponent');
  });
});
