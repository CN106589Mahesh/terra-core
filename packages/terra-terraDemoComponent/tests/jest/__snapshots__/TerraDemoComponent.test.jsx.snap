// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TerraDemoComponent should render a default component 1`] = `
<button
  className="terraDemoComponent"
>
  default
</button>
`;
